crates/matrix-sdk-ui/src/sync_service.rs

```diff
@@ -360,6 +469,17 @@ impl SyncServiceInner {
 
         Ok(())
     }
+
+    async fn restart(
```

Comment:
Since this is called only once and the body is trivial, might be worth inlining? (unless it's used in more places in subsequent commits)

Reply:
I prefer the semantics of a `Offline -> restart()` call instead of having to read that a `stop()` in combination of a `start()` is a restart.

I'll let the compiler inline this.